feat: Add toggle for creating non-public playlists

TechGeekGamer opened the pull request:
This allows for making playlists limited to the current user (toggled off), rather than shared with all logged in users (toggled on, default) 

This feature requires [Jellyfin Server 10.9.0](https://github.com/jellyfin/jellyfin/releases/tag/v10.9.0) or higher. I tested this on my own Jellyfin instance running system version 10.9.9.

![simulator_screenshot_6F7F067E-961F-4526-BA9C-18F30E532677](https://github.com/user-attachments/assets/34b6d163-7197-4972-96df-14859a41e4a7)

hax4dazy reviewed (approved):
LGTM!

rasmuslos commented:
Does this break compatibility with 10.8.x instances?

rasmuslos commented:
Sorry for the late reply! I was wondering if this breaks compatibility with 10.8.x servers?

TechGeekGamer commented:
No problem! I haven't actually tested that, I'll do that and report back

TechGeekGamer commented:
Hey @rasmuslos, I tested it and it does not _appear_ to break on 10.8.x server versions (tested on 10.8.13), however I made a small change so the toggle becomes unavailable on versions less than 10.9.0- do you think there is a better way to handle this?

<img width="295" alt="screenshot of add playlist panel with 'share with instance' toggle disabled" src="https://github.com/user-attachments/assets/f59d53a3-36b5-4771-8d35-451341906c4e">

rasmuslos commented:
I added a feature flag system, you can check for availability using `JellyfinClient.shared.supports(.sharedPlaylists)`

rasmuslos commented on AmpFinKit/Sources/AFNetwork/Methods/JellyfinClient+Playlist.swift:
```diff
@@ -16,11 +16,12 @@ private let playlistQuery = [
 
 public extension JellyfinClient {
-    func create(playlistName: String, trackIds: [String]) async throws {
+    func create(playlistName: String, trackIds: [String], IsPublic: Bool) async throws {
```
This is a minor thing, but swift argument labels should start in lowercase, not uppercase

TechGeekGamer replied:
Resolved in a25ef92db13c33cb1c7ec1cd5f4957bb579f3266

TechGeekGamer commented:
> I added a feature flag system, you can check for availability using `JellyfinClient.shared.supports(.sharedPlaylists)`

@rasmuslos Oh thanks, that's really helpful! Updated the toggle to rely on that. I've also tested the toggle a bit, behaves as expected with it being disabled on 10.8.13 and enabled on 10.9.11.

rasmuslos commented on Multiplatform/Collections/Playlists/PlaylistAddSheet.swift:
```diff
@@ -14,8 +14,11 @@ struct PlaylistAddSheet: View {
     
     let track: Track
     
+    @State private var serverVersion: String?
```
This can be removed

TechGeekGamer replied:
Resolved in 712e3723c0e3dd353082cbd1624092ce1a94dff9

rasmuslos commented on Multiplatform/Collections/Playlists/PlaylistAddSheet.swift:
```diff
@@ -117,6 +124,15 @@ struct PlaylistAddSheet: View {
             #endif
             .toolbarTitleDisplayMode(.inline)
         }
+        .task {
```
This too

rasmuslos commented:
Otherwise it looks good, Thanks!
